### PR TITLE
[WIP] USGS basin scaling for each US 2023 sub GMM

### PR DESCRIPTION
**Pull request by CB-quakemodel**
PR to add the USGS basin scaling adjustments to each of the required subduction GMMs in the 2023 US model.

To do:

- [x] AbrahamsonGulerce2020
- [x] ParkerEtAl2020
- [x] KuehnEtAl2020
- [x] Not applied to AtkinsonMacias2009
- [x] Not applied to ZhaoEtAl2006
- [ ] Unit tests for `utils_usgs_basin_scaling.py`
- [ ] Provide some SMT plots here demonstrating impact of the scalings

### CODE REVIEW

File: openquake/hazardlib/gsim/parker_2020.py [hunk 1]
```diff
@@ -399,6 +400,11 @@ def __init__(self, region=None, saturation_region=None, basin=None,
         else:
             self.saturation_region = saturation_region
         self.basin = basin
+        self.usgs_basin_scaling = usgs_basin_scaling
+        if (self.usgs_basin_scaling and 'z2pt5' not in
+            self.REQUIRES_SITES_PARAMETERS):
+            raise ValueError('User must specify a GSIM class for this GMPE '
```
Comment (micheles) [inline openquake/hazardlib/gsim/parker_2020.py:406]: Maybe the error message came from the ModifiableGMPE? It does not make sense here.

File: openquake/hazardlib/gsim/parker_2020.py [hunk 2]
```diff
@@ -414,6 +420,12 @@ def compute(self, ctx: np.recarray, imts, mean, sig, tau, phi):
         for m, imt in enumerate(imts):
             C = self.COEFFS[imt]
 
+            # USGS basin scaling factor is imt-dependent
+            if self.usgs_basin_scaling:
+                usgs_baf = USGSBasinScaling(ctx.z2pt5, imt)
```
Comment (micheles) [inline openquake/hazardlib/gsim/parker_2020.py:425]: Are you forgetting a call to `get_basin_factor()`?

File: openquake/hazardlib/gsim/utils_usgs_basin_scaling.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    constr = np.clip(z, basin_upper, basin_lower)
+    basin_range = basin_lower - basin_upper
+    z_scale = (constr - basin_upper) / basin_range
+    if period == 0.75:
+        return 0.585 * z_scale
+    else:
+        return z_scale
+    
+
+class USGSBasinScaling(object):
```
Comment (micheles) [inline openquake/hazardlib/gsim/utils_usgs_basin_scaling.py:64]: A class with a single method can be replaced with a function, so it is not clear why you want a class here.

File: openquake/hazardlib/gsim/parker_2020.py (hunk 1 above)
Comment (CB-quakemodel) [reply openquake/hazardlib/gsim/parker_2020.py:406]: The user might specify a  to use the US basin scaling, but specify a class of this GMM which does not consider basin effects (especially for this gsim where there is `ParkerEtAl2020SInterface` and `ParkerEtAl2020SInterB`, and so I think a warning might be a good idea to highlight this?

**Pull request closed without merging**